src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -177,4 +177,43 @@ public void delete(@PathVariable Long id) {
         List<Book> bookList = bookRepository.findByPublisherAndGenreAndYear(publisher, genre, year);
         return new ResponseEntity<>(bookList, HttpStatus.OK);
     }
+
+    /**
+     * Method to search a book by all parameters
+     *
+     * @param id        parameter to search by id
+     * @param genre     parameter to search by genre
+     * @param author    parameter to search by author
+     * @param image     parameter to search by image
+     * @param title     parameter to search by title
+     * @param subtitle  parameter to search by subtitle
+     * @param publisher parameter to search by publisher
+     * @param startYear parameter to filter by year (initial)
+     * @param endYear   parameter to filter by year  (final)
+     * @param pages     parameter to search by pages
+     * @param isbn      parameter to search by isbn
+     * @return book depending on the parameters
+     */
+    @ApiOperation(value = "Method to search a book by all parameters", response = Book.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Book found successfully"),
+            @ApiResponse(code = 404, message = "Book not found")
+    })
+    @GetMapping("/parameters")
```

Comment:
Maybe you should use the index endpoint and add this parameters

Reply:
Se desestima comentario para dejar de manera explicita las distintas maneras de hacer queries en los repositorios